src/js/actions/deploymentActions.js

```diff
@@ -113,28 +113,41 @@ export const getSingleDeploymentStats = id => dispatch =>
     dispatch({ type: DeploymentConstants.RECEIVE_DEPLOYMENT_STATS, stats: res.data, deploymentId: id })
   );
 
-export const getSingleDeployment = id => (dispatch, getState) =>
-  Promise.all([GeneralApi.get(`${deploymentsApiUrl}/deployments/${id}`), GeneralApi.get(`${deploymentsApiUrl}/deployments/${id}/devices`)]).then(responses => {
-    const [deploymentRes, deploymentDeviceListRes] = responses;
-    const storedDeployment = getState().deployments.byId[id] || {};
-    const deploymentDevices = storedDeployment.devices || {};
-    const devices = deploymentDeviceListRes.data.reduce((accu, item) => {
+export const getDeploymentDevices = (id, options = {}) => (dispatch, getState) => {
+  const { page = defaultPage, perPage = defaultPerPage } = options;
+  return GeneralApi.get(`${deploymentsApiUrl}/deployments/${id}/devices/list?deployment_id=${id}&page=${page}&per_page=${perPage}`).then(response => {
```

Comment:
From the backend PoV, there is no need to pass in thee `deployment_id` GET parameter.
Why are you sending it?

Reply:
oops, I think I misread the docs there and treated the path parameter as a query parameter 😞